store/deposit-dates.js

```diff
@@ -45,7 +46,7 @@ class DepositDates {
     return Math.round(level * 100) / 100
   }
 
-  async retrieveDepositDates(pageNumber, searchTerm, columnOrder) {
+  retrieveDepositDates = async (pageNumber, searchTerm, columnOrder) => {
```

Comment:
Because `this` was undefined otherwise. I could have bind `this` in the constructor but I found this more readable.

Reply:
Where and why? You always call it using dot.notation, this should be defined everywhere. I consider this being bad practice everywhere except react components.

Reply:
https://github.com/oacore/dashboard/blob/bug-fixes/pages/data-providers/%5Bdata-provider-id%5D/content.jsx#L114


It looks like when a function is passed as a prop it loses this.

Reply:
Yes it does and it should. You have to do something like `(...args) => store.works.retrieveWorks(...args)` or `store.works.retrieveWorks.bind(store.works)` but linter does not like the second one.

Reply:
Isn't it the same? I mean arrow function is bulletproof. 

Doing `(...args) => store.works.retrieveWorks(...args)` may cause bugs, because it's not obvious on first sight. If you insist I will change it.

Reply:
Everybody likes doing `hasKey = Object.prototype.hasOwnKey.call`. Let's keep JavaScript being JavaScript and do not bind methods by default. It might never happen in this project but I assume that a JS developer expects prototype be similar to usual JS prototype.

Which bugs? It's the same bind but in the other place. All it does – redirects arguments.

Reply:
> Which bugs? It's the same bind but in the other place. All it does – redirects arguments.

Yes but in that other place, you do it only once and not care about it at all in the whole codebase.


Anyway, I am not going to "fight" over such small detail. All I am trying to point out is that if we use `(...args) => store.works.retrieveWorks(...args)` there is a tradeoff creating new function on every render. (To be honest, I don't care about it much in small components but <Table /> is quite a big component) Yes, it can be solved by creating class method but it brings more code just for nothing.